### ISSUE TITLE
Feature Request: lockable tabs/files

### ISSUE THREAD

**Issue by AtmanActive** [Komodo/KomodoEdit#638]
Mmmmmm, I'm dreaming here... it would be really nice if I could right click on a tab and then select the option 'Lock this tab' which would make it not close when I choose File->Close->All Files and which would not change it's position on various tabs-sorting functions.

Just wanted to share my dream. :-)

Thank you.

**Comment by cgchoffman** [comment]
Huh, I could see this being handy.  I've never thought about it but it could make it quick to "context switch".  You could accomplish this by using the Custom Workspaces.  Have you tried using those yet?  The idea being that the files you would have locked would be open in your "saved workspace" then when you want to clear the deck you would *close all* then *File > Open > Workspace*.

**Comment by AtmanActive** [comment]
Thank you very much for the tip.
I will try to play around with workspaces, I didn't have much time yet, and I believe they are the same thing that FireFox's TabPlus had, like save the entire Komodo's running state, which is nice, of course.

But even then, in some working environments, there are simply some files that you just keep on opening and closing forever and then you realize that it would be best to have it open all the time, like some primary 'conf' file or something similar that you need on daily basis.
Frankly, I open and close around 20 files a day and soon enough that one 'conf' file is lost from 'recently used list' and then I need to find it manually, reopen it again, then of course, it is open at the end of all tabs, but want to have it first as a first tab, then I need to drag and drop it through all open tabs, and I just find myself doing that every few days over and over again. :-)

Now, workspaces could potentially do that, like, I save the workspace where I have those few essential files already open, and then I work from there, opening and closing other files through the day. Then, when I'm finished with that dev cycle, I would go 'close all' and then 'open that particular workspace'.

I'll try that approach in the following days for sure.

Thank you.

**Comment by cgchoffman** [comment]
Yep, I'm not closing this.  It's an interesting idea.  Not sure if anyone else does it.  But it won't be implemented any time soon so might as well try using Custom Workspaces in the mean time.

**Comment by cdcmicro** [comment]
A previous version of Komodo IDE (v6 or v7) allowed me to drop a file onto the toolbox.  I kind of miss it. The use case here sounds similar.

**Comment by AtmanActive** [comment]
Yup, having a one-click-file-open shortcut somewhere on the UI would help here as well.

**Comment by cgchoffman** [comment]
> A previous version of Komodo IDE (v6 or v7) allowed me to drop a file onto the toolbox. I kind of miss it. The use case here sounds similar.

I don't think you commented on the right ticket.  Not sure how your comments relates to locking tabs.  This functionality exists and is working as far as I know so if it's not working for you please open a [forum post](http://forum.komodoide.com/) and we'll get you sorted.  It may result in another ticket being opened but for now this should be working.

> Yup, having a one-click-file-open shortcut somewhere on the UI would help here as well.

Please open another ticket and try to keep existing tickets on topic.

**Comment by AtmanActive** [comment]
@cgchoffman, your assessment here is not correct.

I just tried dragging and dropping a file on my right-pane toolbox and got: 
"Internal error: Komodo doesn't know how to drag/drop file:..."
So, no, this functionality does not exist or is not working at all.

The idea of locking tabs serves to make a few files 'special' and handy at all times. Having such a file on the toolbox/toolbar could also make it 'special' and handy at all times.
So, yes, that is exactly why these comments are on the right ticket.
We are simply suggesting workarounds for devs and users alike, but it seems that it is OK when you are doing the same thing, but not OK when others try to do it.

**Comment by cdcmicro** [comment]
So, yes, that is exactly why these comments are on the right ticket.

Thank you, @AtmanActive  

@cgchoffman, dropping a file shortcut on the toolbox hasn't worked for me for a couple of major releases.  If will open a discussion on the forums.  Maybe it's working as expected.  Maybe not.

**Comment by cdcmicro** [comment]
Opened forum topic 2091 (File shortcut in Toolbox)

**Comment by cgchoffman** [comment]
Hey guys, I apologize for the tone.  I'm simply trying to keep the ticket focused on the request.

@cdcmicro I misunderstood your comment, I thought you either meant adding a tool to the toolbox or having a file open in the editor by dropping it on the toolbox (the former being possible, the latter not possible at the moment).  I responded in the [ forum post](http://forum.komodoide.com/t/file-shortcut-in-toolbox/2091)  If it existed then this should be opened in a regression bug, not here.  If it worked it would be a great suggested workaround.

@AtmanActive I did not follow that you were suggesting work around for the request here.  It looked like another enhancement request within the enhencement request which should have another ticket.  You could make a macro that opens a file then add that to your tool bar if you like.  Sorry to have dismissed that as a workaround.  It sounds like @cdcmicro can use this too though it's not as convenient.

```
ko.open.URI("path/to/file")
```

**Comment by AtmanActive** [comment]
> ko.open.URI("path/to/file")

Yes, that will do for now.

Thank you very much.

**Comment by cdcmicro** [comment]
>ko.open.URI("path/to/file")

Yup.  The macro works for me, too.  Thanks very much.
